crates/sui-package-resolver/Cargo.toml

```diff
@@ -0,0 +1,43 @@
+[package]
+name = "sui-package-resolver"
+version = "0.1.0"
+edition = "2021"
+authors = ["Mysten Labs <build@mystenlabs.com>"]
+license = "Apache-2.0"
+publish = false
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+async-trait.workspace = true
+bcs.workspace = true
+move-binary-format.workspace = true
+move-core-types.workspace = true
+sui-types.workspace = true
+thiserror.workspace = true
+sui-rest-api.workspace = true
+tokio.workspace = true
+eyre.workspace = true
+serde.workspace = true
+lru.workspace = true
+typed-store-derive = { optional = true, workspace = true }
+typed-store = { optional = true, workspace = true }
+rocksdb = { optional = true, workspace = true }
+tempfile = { optional = true, workspace = true }
+diesel = { optional = true, workspace = true }
+sui-indexer = { optional = true, workspace = true }
+workspace-hack.workspace = true
+
+[dev-dependencies]
+expect-test.workspace = true
+hyper.workspace = true
+insta.workspace = true
+move-compiler.workspace = true
+serde_json.workspace = true
+sui-move-build.workspace = true
+tower.workspace = true
+
+[features]
+pg_integration = ["dep:diesel", "dep:sui-indexer"]
+rocksdb_integration = ["dep:rocksdb", "dep:tempfile", "dep:typed-store-derive", "dep:typed-store"]
```

Comment:
I think per the conversation here we wanted to have these implementations live in separate crates or where they're being use. https://github.com/MystenLabs/sui/pull/14565#discussion_r1383550672

Reply:
Moved the `package_store` implementations to where they are being used